eftest/src/eftest/runner.clj

```diff
@@ -99,29 +99,35 @@
         (once-fixtures
           (fn []
             (if (multithread-vars? opts)
-              (do (->> vars (filter synchronized?) (map test-var)   (dorun))
-                  (->> vars (remove synchronized?) (pmap* test-var) (dorun)))
+              (do (->> vars (filter synchronized?) (map test-var) (dorun))
+                  (->> vars (remove synchronized?) (pmap* executor test-var) (dorun)))
               (doseq [v vars] (test-var v)))))
         (catch Throwable t
           (test/do-report (fixture-exception t)))))))
 
-(defn- test-ns [ns vars report opts]
+(defn- test-ns [ns vars executor report opts]
   (let [ns (the-ns ns)]
     (binding [test/*report-counters* (ref test/*initial-report-counters*)]
       (test/do-report {:type :begin-test-ns, :ns ns})
-      (test-vars ns vars report opts)
+      (test-vars ns vars executor report opts)
       (test/do-report {:type :end-test-ns, :ns ns})
       @test/*report-counters*)))
 
 (defn- test-all [vars {:as opts :keys [capture-output?] :or {capture-output? true}}]
   (let [report (synchronize test/report)
-        mapf   (if (multithread-namespaces? opts) pmap* map)
+        executor (threadpool-executor opts)
```

Comment:
Align let forms, and it's probably better to define the executor in `run-tests`, considering that `:executor` might be an option in future.